api/utils/endpoints.go

```diff
@@ -1,57 +1,60 @@
 package utils
 
 import (
-	"fmt"
 	"log"
+	"net"
 	"net/url"
-	"path"
 	"strconv"
+	"strings"
 )
 
+const apiPrefix = "/api"
+
 func ApiListenUrl() *url.URL {
+	const defaultIP = "127.0.0.1"
 	const defaultPort = "4001"
 
-	shouldServeUI := ShouldServeUI()
-
-	apiPrefix := "/"
-	if shouldServeUI {
-		apiPrefix = "/api"
+	// Reuse the normalization logic so absolute/relative values are both supported.
+	// If EnvAPIEndpoint contains a full URL, use its path component.
+	apiPath := ApiEndpointUrl().Path
+	if apiPath == "" {
```

Comment:
I see this logic is repeated [below](https://github.com/photoview/photoview/pull/1311/files#diff-a1decf6e4146e7454952c2a5dabb498b8bed149be1c12a16cf2d7e062038b8d2R55). Since `ApiEndpointUrl()` is exposed, I suggest to remove this check and rely on `ApiEndpointUrl()` to handle the default value.